docs/examples/user-configuration/user-config-template.yaml

```diff
@@ -1,145 +1,212 @@
-      selector:
-        deploymentConfig: ${APPLICATION_NAME}
-  - apiVersion: v1
-    kind: Service
-    metadata:
-      annotations:
-        description: The JGroups ping port for clustering.
-        service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
-      labels:
-        application: ${APPLICATION_NAME}
-      name: ${APPLICATION_NAME}-ping
-    spec:
-      clusterIP: None
-      ports:
-        - name: ping
-          port: 8888
-      selector:
-        deploymentConfig: ${APPLICATION_NAME}
-  - apiVersion: v1
-    kind: DeploymentConfig
-    metadata:
-      labels:
-        application: ${APPLICATION_NAME}
-      name: ${APPLICATION_NAME}
-    spec:
-      replicas: 1
-      selector:
-        deploymentConfig: ${APPLICATION_NAME}
-      strategy:
-        type: Recreate
-      template:
-        metadata:
-          labels:
-            application: ${APPLICATION_NAME}
-            deploymentConfig: ${APPLICATION_NAME}
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    name: ${APPLICATION_NAME}
+    labels:
+      application: ${APPLICATION_NAME}
+  stringData:
+    application-password: ${APPLICATION_PASSWORD}
+    application-user: ${APPLICATION_USER}
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      description: Provides a ping service for clustered applications.
+      service.alpha.kubernetes.io/tolerate-unready-endpoints: "true"
+    labels:
+      application: ${APPLICATION_NAME}
+    name: ${APPLICATION_NAME}-ping
+  spec:
+    clusterIP: None
+    ports:
+    - name: ping
+      port: 8888
+    selector:
+      deploymentConfig: ${APPLICATION_NAME}
+- apiVersion: v1
+  kind: Service
+  metadata:
+    annotations:
+      description: Provides a service for accessing the application over HTTPS or Hot Rod protocol.
+      service.alpha.openshift.io/serving-cert-secret-name: service-certs
+    labels:
+      application: ${APPLICATION_NAME}
+    name: ${APPLICATION_NAME}
+  spec:
+    ports:
+    - name: hotrod
+      port: 11222
+      targetPort: 11222
+    - name: https
+      port: 8443
+      targetPort: 8443
+    selector:
+      deploymentConfig: ${APPLICATION_NAME}
+- apiVersion: apps/v1beta1
+  kind: StatefulSet
+  metadata:
+    labels:
+      application: ${APPLICATION_NAME}
+    name: ${APPLICATION_NAME}
+  spec:
+    replicas: ${{NUMBER_OF_INSTANCES}}
+    serviceName: ${APPLICATION_NAME}-ping
+    strategy:
+      rollingParams:
+        intervalSeconds: 20
+        maxSurge: 1
+        maxUnavailable: 1
+        timeoutSeconds: 1200
+        updatePeriodSeconds: 20
+      type: Rolling
+    template:
+      metadata:
+        labels:
+          application: ${APPLICATION_NAME}
+          deploymentConfig: ${APPLICATION_NAME}
+        name: ${APPLICATION_NAME}
+      spec:
+        containers:
+        - env:
+          - name: USER_CONFIG_MAP
+            value: "true"
+          - name: SERVICE_NAME
+            value: ${APPLICATION_NAME}
+          - name: SERVICE_PROFILE
```

Comment:
`SERVICE_NAME` and `SERVICE_PROFILE` vars don't make sense with the user config as they'll never be used.